# Déjame escuchar la música

# Contenido <a id='back'></a>

* [Introducción](#intro)
* [Etapa 1. Descripción de los datos](#data_review)
    * [Conclusiones](#data_review_conclusions)
* [Etapa 2. Preprocesamiento de datos](#data_preprocessing)
    * [2.1 Estilo del encabezado](#header_style)
    * [2.2 Valores ausentes](#missing_values)
    * [2.3 Duplicados](#duplicates)
    * [2.4 Conclusiones](#data_preprocessing_conclusions)
* [Etapa 3. Prueba de hipótesis](#hypothesis)
    * [3.1 Hipótesis 1: actividad de los usuarios y las usuarias en las dos ciudades](#activity)
* [Conclusiones](#end)

## Introducción <a id='intro'></a>
Como analista de datos, tu trabajo consiste en analizar datos para extraer información valiosa y tomar decisiones basadas en ellos. Esto implica diferentes etapas, como la descripción general de los datos, el preprocesamiento y la prueba de hipótesis.

Siempre que investigamos, necesitamos formular hipótesis que después podamos probar. A veces aceptamos estas hipótesis; otras veces, las rechazamos. Para tomar las decisiones correctas, una empresa debe ser capaz de entender si está haciendo las suposiciones correctas.

En este proyecto, compararás las preferencias musicales de las ciudades de Springfield y Shelbyville. Estudiarás datos reales de transmisión de música online para probar la hipótesis a continuación y comparar el comportamiento de los usuarios y las usuarias de estas dos ciudades.

### Objetivo:
Prueba la hipótesis:
1. La actividad de los usuarios y las usuarias difiere según el día de la semana y dependiendo de la ciudad.


### Etapas
Los datos del comportamiento del usuario se almacenan en el archivo `/datasets/music_project_en.csv`. No hay ninguna información sobre la calidad de los datos, así que necesitarás examinarlos antes de probar la hipótesis.

Primero, evaluarás la calidad de los datos y verás si los problemas son significativos. Entonces, durante el preprocesamiento de datos, tomarás en cuenta los problemas más críticos.

Tu proyecto consistirá en tres etapas:
 1. Descripción de los datos.
 2. Preprocesamiento de datos.
 3. Prueba de hipótesis.








[Volver a Contenidos](#back)

## Etapa 1. Descripción de los datos <a id='data_review'></a>

Abre los datos y examínalos.

Necesitarás `pandas`, así que impórtalo.

In [1]:
# Importar pandas
import pandas as pd

Lee el archivo `music_project_en.csv` de la carpeta `/datasets/` y guárdalo en la variable `df`:

In [2]:
# Leer el archivo y almacenarlo en df
#Debido a que estoy haciendo el proyecto la herramienta externa Kaggle, la ruta será diferente. 
music_df = pd.read_csv('../Data/music_project_en.csv')


Muestra las 10 primeras filas de la tabla:

In [3]:
# Obtener las 10 primeras filas de la tabla df
music_df.head(10)


userID                        Track            artist   genre  \
0  FFB692EC            Kamigata To Boots  The Mass Missile    rock   
1  55204538  Delayed Because of Accident  Andreas Rönnberg    rock   
2    20EC38            Funiculì funiculà       Mario Lanza     pop   
3  A3DD03C9        Dragons in the Sunset        Fire + Ice    folk   
4  E2DC1FAE                  Soul People        Space Echo   dance   
5  842029A1                       Chains          Obladaet  rusrap   
6  4CB90AA5                         True      Roman Messer   dance   
7  F03E1C1F             Feeling This Way   Polina Griffith   dance   
8  8FA1D3BE                     L’estate       Julia Dalia  ruspop   
9  E772D5C0                    Pessimist               NaN   dance   

        City        time        Day  
0  Shelbyville  20:28:33  Wednesday  
1  Springfield  14:07:09     Friday  
2  Shelbyville  20:58:07  Wednesday  
3  Shelbyville  08:37:09     Monday  
4  Springfield  08:34:34     Monday  
5  Shelbyville  13:09:41     Friday  
6  Springfield  13:00:07  Wednesday  
7  Springfield  20:47:49  Wednesday  
8  Springfield  09:17:40     Friday  
9  Shelbyville  21:20:49  Wednesday

Obtén la información general sobre la tabla con un comando. Conoces el método que muestra la información general que necesitamos.

In [4]:
# Obtener la información general sobre nuestros datos
music_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65079 entries, 0 to 65078
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0     userID  65079 non-null  object
 1   Track     63736 non-null  object
 2   artist    57512 non-null  object
 3   genre     63881 non-null  object
 4     City    65079 non-null  object
 5   time      65079 non-null  object
 6   Day       65079 non-null  object
dtypes: object(7)
memory usage: 3.5+ MB


Estas son nuestras observaciones sobre la tabla. Contiene siete columnas. Almacenan los mismos tipos de datos: `object`.

Según la documentación:
- `' userID'`: identificador del usuario o la usuaria;
- `'Track'`: título de la canción;
- `'artist'`: nombre del artista;
- `'genre'`: género de la pista;
- `'City'`: ciudad del usuario o la usuaria;
- `'time'`: la hora exacta en la que se reprodujo la canción;
- `'Day'`: día de la semana.

Podemos ver tres problemas con el estilo en los encabezados de la tabla:
1. Algunos encabezados están en mayúsculas, otros en minúsculas.
2. Hay espacios en algunos encabezados.
3. `Detecta el tercer problema por tu cuenta y descríbelo aquí`.
4. El titulo de la columna 'Time' no es muy descriptivo, podría cambiarlo por algo como 'total_play' 



### Escribe observaciones de tu parte. Estas son algunas de las preguntas que pueden ser útiles: <a id='data_review_conclusions'></a>

`1.   ¿Qué tipo de datos tenemos a nuestra disposición en las filas? ¿Y cómo podemos entender lo que almacenan las columnas?`
Para empezar observamos que tenemos un DataFrame de 65079 filas x 7 columnas, donde la mayoría de estas columnas NO son cuantitativas, expresan datos categóricos o característicos. A continuación explico brevemente que contiene cada columna.
* La columna user_id contiene el identificador único para cada usuario.
* La columna track contiene el nombre de la canción que reprodujo.
* La columna artist muestra el nombre del artista de dicha canción.
* La columna genre muestra el nombre del genero de la canción que el usuario reprodujo.
* La columna city muestra la ciudad en la que el usuario escucho la canción, Springfield o Shelbyville.
* La columna time muestra el tiempo de reproducción que el usuario reprodujo esa canción en horas, minutos y segundos.
* La columna day muestra el día de la semana en que el usuario reprodujo la canción.


`2.   ¿Hay suficientes datos para proporcionar respuestas a nuestra hipótesis o necesitamos más información?`
Considero que si hay suficientes datos para aceptar o negar la hipótesis propuesta ya que tenemos el día de la semana en que la canción se reprodujo y también la ciudad, adicionalmente contamos con información detallada de la reproducción. Cabe recalcar que esta información es suficiente para aceptar o negar la hipótesis de manera ambigua, debido a que existen muchos factores y variables distintas que influyen en la actividad de un usuario, no solo el día de la semana y la ciudad,un ejemplo serían las circunstancias por las que el usuario pasa. Sin embargo, mi respuesta es si, hay suficiente información para dar respuesta a la hipótesis.
`3.   ¿Notaste algún problema en los datos, como valores ausentes, duplicados o tipos de datos incorrectos?`
Si. Para empezar los nombres de las columnas tienen distintos errores, deberíamos estandarizarlos. Después con el método info podemos ver que las columnas track, artist y genre tienen valores ausentes, los cuales hay que tratar. Por último la columna de time es de tipo objeto, y aunque probablemente no necesitemos realizar operaciones con ella, si podríamos hacer operaciones, y considero que es importante si se tiene la posibilidad el tener la capacidad. Entonces deberíamos cambiarla a tipo timedelta con pandas.

Después deberíamos buscar valores duplicados explícitos e implícitos para saber y tratarlos, creo que de momento eso serían todos los problemas de calidad de los datos. 

[Volver a Contenidos](#back)

## Etapa 2. Preprocesamiento de datos <a id='data_preprocessing'></a>

El objetivo aquí es preparar los datos para que sean analizados.
El primer paso es resolver cualquier problema con los encabezados. Luego podemos avanzar a los valores ausentes y duplicados. Empecemos.

Corrige el formato en los encabezados de la tabla.


### Estilo del encabezado <a id='header_style'></a>
Muestra los encabezados de la tabla (los nombres de las columnas):

In [5]:
music_df.columns


Index(['  userID', 'Track', 'artist', 'genre', '  City  ', 'time', 'Day'], dtype='object')

Cambia los encabezados de la tabla de acuerdo con las reglas del buen estilo:
* Todos los caracteres deben ser minúsculas.
* Elimina los espacios.
* Si el nombre tiene varias palabras, utiliza snake_case.

Anteriormente, aprendiste acerca de la forma automática de cambiar el nombre de las columnas. Vamos a aplicarla ahora. Utiliza el bucle for para iterar sobre los nombres de las columnas y poner todos los caracteres en minúsculas. Cuando hayas terminado, vuelve a mostrar los encabezados de la tabla:

Ahora, utilizando el mismo método, elimina los espacios al principio y al final de los nombres de las columnas e imprime los nombres de las columnas nuevamente:

Necesitamos aplicar la regla de snake_case a la columna `userid`. Debe ser `user_id`. Cambia el nombre de esta columna y muestra los nombres de todas las columnas cuando hayas terminado.

In [6]:
# Bucle en los encabezados poniendo todo en minúsculas
# Bucle en los encabezados eliminando los espacios
# Cambiar el nombre de la columna "userid"
#Para motivos de sencillez elaborare todo en el mismo bucle y en la misma celda.
new_cols = []
for col in music_df.columns:
    name_stripped = col.strip()
    name_lowered = name_stripped.lower()
    snake_case = name_lowered.replace(' ', '_')
    new_cols.append(snake_case)
music_df.columns = new_cols



Comprueba el resultado. Muestra los encabezados una vez más:

In [7]:
# Comprobar el resultado: la lista de encabezados
music_df.columns

Index(['userid', 'track', 'artist', 'genre', 'city', 'time', 'day'], dtype='object')

[Volver a Contenidos](#back)

### Valores ausentes <a id='missing_values'></a>
 Primero, encuentra el número de valores ausentes en la tabla. Debes utilizar dos métodos en una secuencia para obtener el número de valores ausentes.

In [8]:
# Calcular el número de valores ausentes
#A través de la siguiente linea de código podemos visualizar el número de valores ausentes para cada columna
music_df.isna().sum()

userid       0
track     1343
artist    7567
genre     1198
city         0
time         0
day          0
dtype: int64

No todos los valores ausentes afectan a la investigación. Por ejemplo, los valores ausentes en `track` y `artist` no son cruciales. Simplemente puedes reemplazarlos con valores predeterminados como el string `'unknown'` (desconocido).

Pero los valores ausentes en `'genre'` pueden afectar la comparación entre las preferencias musicales de Springfield y Shelbyville. En la vida real, sería útil saber las razones por las cuales hay datos ausentes e intentar recuperarlos. Pero no tenemos esa oportunidad en este proyecto. Así que tendrás que:
* rellenar estos valores ausentes con un valor predeterminado;
* evaluar cuánto podrían afectar los valores ausentes a tus cómputos;

Reemplazar los valores ausentes en las columnas `'track'`, `'artist'` y `'genre'` con el string `'unknown'`. Como mostramos anteriormente en las lecciones, la mejor forma de hacerlo es crear una lista que almacene los nombres de las columnas donde se necesita el reemplazo. Luego, utiliza esta lista e itera sobre las columnas donde se necesita el reemplazo haciendo el propio reemplazo.

In [9]:
music_df['track'] = music_df['track'].fillna('unknown')
music_df['artist'] = music_df['artist'].fillna('unknown')
music_df['genre'] = music_df['genre'].fillna('unknown')

Ahora comprueba el resultado para asegurarte de que después del reemplazo no haya valores ausentes en el conjunto de datos. Para hacer esto, cuenta los valores ausentes nuevamente.

In [10]:
# Contar valores ausentes
music_df.isna().sum()

userid    0
track     0
artist    0
genre     0
city      0
time      0
day       0
dtype: int64

[Volver a Contenidos](#back)

### Duplicados <a id='duplicates'></a>
Encuentra el número de duplicados explícitos en la tabla. Una vez más, debes aplicar dos métodos en una secuencia para obtener la cantidad de duplicados explícitos.

In [11]:
# Contar duplicados explícitos
music_df.duplicated().sum()

3826

Ahora, elimina todos los duplicados. Para ello, llama al método que hace exactamente esto.

In [12]:
# Eliminar duplicados explícitos
music_df.drop_duplicates(inplace = True)

Comprobemos ahora si eliminamos con éxito todos los duplicados. Cuenta los duplicados explícitos una vez más para asegurarte de haberlos eliminado todos:

In [13]:
# Comprobar de nuevo si hay duplicados

music_df.duplicated().sum()

0

Ahora queremos deshacernos de los duplicados implícitos en la columna `genre`. Por ejemplo, el nombre de un género se puede escribir de varias formas. Dichos errores también pueden afectar al resultado.

Para hacerlo, primero mostremos una lista de nombres de género únicos, ordenados en orden alfabético. Para ello:
* Extrae la columna `genre` del DataFrame.
* Llama al método que devolverá todos los valores únicos en la columna extraída.


In [14]:
# Inspeccionar los nombres de géneros únicos

print(sorted(music_df['genre'].unique()))



['acid', 'acoustic', 'action', 'adult', 'africa', 'afrikaans', 'alternative', 'ambient', 'americana', 'animated', 'anime', 'arabesk', 'arabic', 'arena', 'argentinetango', 'art', 'audiobook', 'avantgarde', 'axé', 'baile', 'balkan', 'beats', 'bigroom', 'black', 'bluegrass', 'blues', 'bollywood', 'bossa', 'brazilian', 'breakbeat', 'breaks', 'broadway', 'cantautori', 'cantopop', 'canzone', 'caribbean', 'caucasian', 'celtic', 'chamber', 'children', 'chill', 'chinese', 'choral', 'christian', 'christmas', 'classical', 'classicmetal', 'club', 'colombian', 'comedy', 'conjazz', 'contemporary', 'country', 'cuban', 'dance', 'dancehall', 'dancepop', 'dark', 'death', 'deep', 'deutschrock', 'deutschspr', 'dirty', 'disco', 'dnb', 'documentary', 'downbeat', 'downtempo', 'drum', 'dub', 'dubstep', 'eastern', 'easy', 'electronic', 'electropop', 'emo', 'entehno', 'epicmetal', 'estrada', 'ethnic', 'eurofolk', 'european', 'experimental', 'extrememetal', 'fado', 'film', 'fitness', 'flamenco', 'folk', 'folklor

Busca en la lista para encontrar duplicados implícitos del género `hiphop`. Estos pueden ser nombres escritos incorrectamente o nombres alternativos para el mismo género.

Verás los siguientes duplicados implícitos:
* `hip`
* `hop`
* `hip-hop`

Para deshacerte de ellos, crea una función llamada `replace_wrong_genres()` con dos parámetros:
* `wrong_genres=`: esta es una lista que contiene todos los valores que necesitas reemplazar.
* `correct_genre=`: este es un string que vas a utilizar como reemplazo.

Como resultado, la función debería corregir los nombres en la columna `'genre'` de la tabla `df`, es decir, remplazar cada valor de la lista `wrong_genres` por el valor en `correct_genre`.

Dentro del cuerpo de la función, utiliza un bucle `'for'` para iterar sobre la lista de géneros incorrectos, extrae la columna `'genre'` y aplica el método `replace` para hacer correcciones.

In [15]:
# Función para reemplazar duplicados implícitos
def replace_wrong_genres(wrong_genres, correct_genre):
    for genre in wrong_genres:
        music_df['genre'] = music_df['genre'].replace(genre, correct_genre)


Ahora, llama a `replace_wrong_genres()` y pásale tales argumentos para que retire los duplicados implícitos (`hip`, `hop` y `hip-hop`) y los reemplace por `hiphop`:

In [16]:
# Eliminar duplicados implícitos
replace_wrong_genres(['hip', 'hop', 'hip-hop'], 'hiphop')

Asegúrate de que los nombres duplicados han sido eliminados. Muestra la lista de valores únicos de la columna `'genre'` una vez más:

In [17]:
# Comprobación de duplicados implícitos
print(sorted(music_df['genre'].unique()))

['acid', 'acoustic', 'action', 'adult', 'africa', 'afrikaans', 'alternative', 'ambient', 'americana', 'animated', 'anime', 'arabesk', 'arabic', 'arena', 'argentinetango', 'art', 'audiobook', 'avantgarde', 'axé', 'baile', 'balkan', 'beats', 'bigroom', 'black', 'bluegrass', 'blues', 'bollywood', 'bossa', 'brazilian', 'breakbeat', 'breaks', 'broadway', 'cantautori', 'cantopop', 'canzone', 'caribbean', 'caucasian', 'celtic', 'chamber', 'children', 'chill', 'chinese', 'choral', 'christian', 'christmas', 'classical', 'classicmetal', 'club', 'colombian', 'comedy', 'conjazz', 'contemporary', 'country', 'cuban', 'dance', 'dancehall', 'dancepop', 'dark', 'death', 'deep', 'deutschrock', 'deutschspr', 'dirty', 'disco', 'dnb', 'documentary', 'downbeat', 'downtempo', 'drum', 'dub', 'dubstep', 'eastern', 'easy', 'electronic', 'electropop', 'emo', 'entehno', 'epicmetal', 'estrada', 'ethnic', 'eurofolk', 'european', 'experimental', 'extrememetal', 'fado', 'film', 'fitness', 'flamenco', 'folk', 'folklor

[Volver a Contenidos](#back)

### Tus observaciones <a id='data_preprocessing_conclusions'></a>

`Para tratar los duplicados explícitos utilizé el método duplicated primeramente para saber cuantos había y posteriormente drop_duplicates para eliminarlos, en general el procesamiento de duplicados explícitos es más rapido que el de duplicados implícitos. Para los duplicados implícitos cree una función que recibia como argumentos la lista de strings incorrectos y el string correcto, dentro se remplaza en el dataframe el valor incorrecto con el valor correcto. Algo que noté es que es mucho más facil identificar valores duplicados implícitos si se ordenan los valores únicos por orden alfabético, de esa manera es mucho más sencillo encontrar errores tipográficos en strings que representan lo mismo.
De igual manera considero que en una tarea del mundo real deberíamos rectificar valroes duplicados implícitos como en la columna City.

[Volver a Contenidos](#back)

## Etapa 3. Prueba de hipótesis <a id='hypothesis'></a>

### Hipótesis: comparar el comportamiento del usuario o la usuaria en las dos ciudades <a id='activity'></a>

La hipótesis afirma que existen diferencias en la forma en que los usuarios y las usuarias de Springfield y Shelbyville consumen música. Para comprobar esto, usa los datos de tres días de la semana: lunes, miércoles y viernes.

* Agrupa a los usuarios y las usuarias por ciudad.
* Compara el número de canciones que cada grupo reprodujo el lunes, el miércoles y el viernes.


Realiza cada cálculo por separado.

El primer paso es evaluar la actividad del usuario en cada ciudad. Recuerda las etapas dividir-aplicar-combinar de las que hablamos anteriormente en la lección. Tu objetivo ahora es agrupar los datos por ciudad, aplicar el método apropiado para contar durante la etapa de aplicación y luego encontrar la cantidad de canciones reproducidas en cada grupo especificando la columna para obtener el recuento.

A continuación se muestra un ejemplo de cómo debería verse el resultado final:
`df.groupby(by='....')['column'].method()`Realiza cada cálculo por separado.

Para evaluar la actividad de los usuarios y las usuarias en cada ciudad, agrupa los datos por ciudad y encuentra la cantidad de canciones reproducidas en cada grupo.



In [18]:
# Contar las canciones reproducidas en cada ciudad
music_df.groupby(by = 'city')['track'].count()

city
Shelbyville    18512
Springfield    42741
Name: track, dtype: int64

`Comenta tus observaciones aquí`

Ahora agrupemos los datos por día de la semana y encontremos el número de canciones reproducidas el lunes, miércoles y viernes. Utiliza el mismo método que antes, pero ahora necesitamos una agrupación diferente.


In [19]:
# Calcular las canciones reproducidas en cada uno de los tres días
music_df.groupby(by='day')['track'].count()


day
Friday       21840
Monday       21354
Wednesday    18059
Name: track, dtype: int64

`Comenta tus observaciones aquí`
La ciudad de Sprinfield escucha más del 100% de la cantidad de canciones que la ciudad de Shelbyville, en cuanto a los días de la semana para las dos ciudades, el día que más canciones se escuchan es el viernes, pero algo interesante es que el número de canciones que escuchan en lunes y viernes es similar, en cambio el día miércoles disminuyen significativamente. En una gráfica esto se vería como un zig-zag, con un declive en el medio.

Ya sabes cómo contar entradas agrupándolas por ciudad o día. Ahora necesitas escribir una función que pueda contar entradas según ambos criterios simultáneamente.

Crea la función `number_tracks()` para calcular el número de canciones reproducidas en un determinado día **y** ciudad. La función debe aceptar dos parámetros:

- `day`: un día de la semana para filtrar. Por ejemplo, `'Monday'` (lunes).
- `city`: una ciudad para filtrar. Por ejemplo, `'Springfield'`.

Dentro de la función, aplicarás un filtrado consecutivo con indexación lógica.

Primero filtra los datos por día y luego filtra la tabla resultante por ciudad.

Después de filtrar los datos por dos criterios, cuenta el número de valores de la columna 'user_id' en la tabla resultante. Este recuento representa el número de entradas que estás buscando. Guarda el resultado en una nueva variable y devuélvelo desde la función.

In [20]:
# Declara la función number_tracks() con dos parámetros: day= y city=.
def number_tracks(day, city):
    df = music_df.query('day == @day and city == @city')
    count = df['userid'].count()
    return count

Llama a `number_tracks()` seis veces, cambiando los valores de los parámetros para que recuperes los datos de ambas ciudades para cada uno de los tres días.

In [21]:
# El número de canciones reproducidas en Springfield el lunes
print('Número de canciones que se reprodujeron en Springfield en lunes: ', number_tracks('Monday', 'Springfield'))

Número de canciones que se reprodujeron en Springfield en lunes:  15740


In [22]:
# El número de canciones reproducidas en Shelbyville el lunes
print('Número de canciones que se reprodujeron en Shelbyville en lunes: ', number_tracks('Monday', 'Shelbyville'))

Número de canciones que se reprodujeron en Shelbyville en lunes:  5614


In [23]:
# El número de canciones reproducidas en Springfield el miércoles
print('Número de canciones que se reprodujeron en Springfield en miércoles: ', number_tracks('Wednesday', 'Springfield'))

Número de canciones que se reprodujeron en Springfield en miércoles:  11056


In [24]:
# El número de canciones reproducidas en Shelbyville el miércoles
print('Número de canciones que se reprodujeron en Shelbyville en miércoles: ', number_tracks('Wednesday', 'Shelbyville'))

Número de canciones que se reprodujeron en Shelbyville en miércoles:  7003


In [25]:
# El número de canciones reproducidas en Springfield el viernes
print('Número de canciones que se reprodujeron en Springfield en viernes: ', number_tracks('Friday', 'Springfield'))

Número de canciones que se reprodujeron en Springfield en viernes:  15945


In [26]:
# El número de canciones reproducidas en Shelbyville el viernes
print('Número de canciones que se reprodujeron en Shelbyville en viernes: ', number_tracks('Friday', 'Shelbyville'))

Número de canciones que se reprodujeron en Shelbyville en viernes:  5895


**Conclusiones**


Primero retomemos la hipótesis: "La actividad de los usuarios y las usuarias difiere según el día de la semana y dependiendo de la ciudad"
Partiendo de esto y juzgando solo esta hipótesis podemos decir que es correcta. ¿Por qué? Por que en los cálculos anteriores se puede observar como claramente las ciudades tienen distintos números de reproducción en cualquier día de la semana que observemos, esto hace correcta la primera implicación de la hipótesis "varia dependiendo de la ciudad" y después también podemos notar que para ambas ciudades el número de reproducciones por cada día de la semana es distinto, para la ciudad de Springfield el lunes y viernes son los días con más reproducciones mientras que el miércoles disminuye, para la ciudad de Shelbyville el día miércoles es cuando más reproducciones hay y los días lunes y viernes son similares en cuanto al número de reproducciones.

[Volver a Contenidos](#back)

# Conclusiones <a id='end'></a>

Springfield:
Para esta ciudad puedo concluir que el número de canciones reproducidas varía según el día de la semana. Los días lunes y viernes son los días en los cuales se escuchan más canciones y el día miércoles este número disminuye significativamente

Shelbyville:
Para esta ciudad de igual manera puedo concluir que el número de canciones que se reproducen varía dependiendo del día de la semana. A diferencia de la ciudad de Springfield, el día que más reproducciones de canciones hay es el miércoles, los días lunes y viernes la cantidad de reproducciones es menor que el miércoles pero entre estos dos días la cantidad es similar.
Springfield y Shelbyville:
El número de canciones que se reproducen es mucho mayor en la ciudad de Springfield que en la de Shelbyville, sin embargo, los datos de las reproducciones se comportan en cierta manera, inversa. Si observaramos este comportamiento en una gráfica de dispersión, los datos de la ciudad de Springfield se verían con los extremos más arriba y el día miércoles que es en medio de la semana más abajo. Por otro lado la gráfica de la ciudad de Shelbyville se vería al reves, los extremos más abajo y el centro, es decir a media semana el día miércoles se vería más arriba debido al número de reproducciones. 
En resumen, para la ciudad de Springfield, los primeros y ultimos días de la semana es cuando más reproducciones tiene, por otro lado en la ciudad de Shelbyville los primeros y ultimos días de la semana son los días en los que menos reproducciones de música hay. 

### Nota
En proyectos de investigación reales, la prueba de hipótesis estadística es más precisa y cuantitativa. También ten en cuenta que no siempre se pueden sacar conclusiones sobre una ciudad entera a partir de datos de una sola fuente.

Aprenderás más sobre la prueba de hipótesis en el sprint de análisis estadístico de datos.

[Volver a Contenidos](#back)